src/Dependencies/Collections/SegmentedList`1.cs

```diff
@@ -0,0 +1,1168 @@
+    // elements. A SegmentedList has a capacity, which is the allocated length
+    // of the internal array. As elements are added to a SegmentedList, the capacity
+    // of the SegmentedList is automatically increased as required by reallocating the
+    // internal array.
+    //
+    [DebuggerTypeProxy(typeof(ICollectionDebugView<>))]
+    [DebuggerDisplay("Count = {Count}")]
+    [Serializable]
+    internal class SegmentedList<T> : IList<T>, IList, IReadOnlyList<T>
+    {
+        private const int DefaultCapacity = 4;
+        private const int MaxArrayLength = 0x7FEFFFFF;
+
+        internal T[] _items; // Do not rename (binary serialization)
+        internal int _size; // Do not rename (binary serialization)
+        private int _version; // Do not rename (binary serialization)
+
+#pragma warning disable CA1825 // avoid the extra generic instantiation for Array.Empty<T>()
+        private static readonly T[] s_emptyArray = new T[0];
+#pragma warning restore CA1825
+
+        // Constructs a SegmentedList. The list is initially empty and has a capacity
+        // of zero. Upon adding the first element to the list the capacity is
+        // increased to DefaultCapacity, and then increased in multiples of two
+        // as required.
+        public SegmentedList()
+        {
+            _items = s_emptyArray;
+        }
+
+        // Constructs a SegmentedList with a given initial capacity. The list is
+        // initially empty, but will have room for the given number of elements
+        // before any reallocations are required.
+        //
+        public SegmentedList(int capacity)
+        {
+            if (capacity < 0)
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity, ExceptionResource.ArgumentOutOfRange_NeedNonNegNum);
+
+            if (capacity == 0)
+                _items = s_emptyArray;
+            else
+                _items = new T[capacity];
+        }
+
+        // Constructs a SegmentedList, copying the contents of the given collection. The
+        // size and capacity of the new list will both be equal to the size of the
+        // given collection.
+        //
+        public SegmentedList(IEnumerable<T> collection)
+        {
+            if (collection == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.collection);
+
+            if (collection is ICollection<T> c)
+            {
+                var count = c.Count;
+                if (count == 0)
+                {
+                    _items = s_emptyArray;
+                }
+                else
+                {
+                    _items = new T[count];
+                    c.CopyTo(_items, 0);
+                    _size = count;
+                }
+            }
+            else
+            {
+                _items = s_emptyArray;
+                using var en = collection!.GetEnumerator();
```

Comment:
>collection! [](start = 31, length = 11)

I'm not sure what this suppression is used for. Can it be removed as part of this cleanup commit (commit 5)?

Reply:
➡️ Now removed